base/Base.jl

```diff
@@ -427,4 +427,11 @@ end
 
 end
 
+# Ensure this file is also tracked
+pushfirst!(_included_files, (Main, abspath(@__FILE__)))
+
 end # baremodule Base
+
+# Set up Main module
+using .Base
+import Base.MainInclude: eval, include
```

Comment:
It'd be really nice if this can somehow be done elsewhere. The idea was to make Base more like a normal package; it should only create the Base module.

Reply:
This can go elsewhere, but I thought it felt the most natural here (it means that if you load just `-J Base.so`, it acts just like you are just inside a fresh `module Main`, instead of being a `baremodule Main`).